Teacher/Teacher/SpeedGrader/MainLayout/View/DrawerContainer.swift

```diff
@@ -58,55 +63,155 @@ struct DrawerContainer<Content: View>: View {
 
     var body: some View {
         VStack(spacing: 0) {
-            Button(action: buttonAction, label: { HStack {
-                Spacer()
-                RoundedRectangle(cornerRadius: 2)
-                    .fill(Color.borderMedium)
-                    .frame(width: 36, height: 4)
-                    .padding(EdgeInsets(top: 24, leading: 0, bottom: 8, trailing: 0))
-                Spacer()
-            } })
-            .padding(.top, -16)
-            .highPriorityGesture(DragGesture(coordinateSpace: .global)
-                .onChanged { value in translation = -value.translation.height }
-                .onEnded { value in withTransaction(DrawerState.transaction) {
-                    let y = height - value.predictedEndTranslation.height - minHeight
-                    let dy = maxHeight - minHeight
-                    state = (y < dy * 0.25) ? .min : (y < dy * 0.75) ? .mid : .max
-                    translation = 0
-                } }
-            )
-            .accessibility(identifier: "SpeedGrader.drawerGripper")
-            .accessibility(label: buttonA11yText)
-            Spacer(minLength: 4)
+            HStack(spacing: 4) {
+                expandCollapseButton
+
+                dragIndicator
+                    .frame(maxWidth: .infinity)
+
+                openCloseButton
+            }
+            .paddingStyle(.horizontal, .standard)
+            .padding(.vertical, 8)
+
             content
         }
-        .frame(maxWidth: 800, maxHeight: max(minHeight, min(maxHeight, height + translation)))
+        .frame(maxHeight: max(minHeight, min(maxHeight, height + translation)))
         .background(DrawerBackground()
             .fill(Color.backgroundLightest)
             .shadow(color: Color.black.opacity(0.3), radius: 3, x: 0, y: 0)
         )
     }
 
-    func buttonAction() { withTransaction(DrawerState.transaction) {
+    @ViewBuilder
+    private var expandCollapseButton: some View {
+        Button(action: expandCollapseButtonAction) {
+            expandCollapseButtonImage
+        }
+        .accessibilityLabel(expandCollapseButtonAccessibilityText)
+        .accessibilityShowsLargeContentViewer {
+            expandCollapseButtonImage
+            expandCollapseButtonAccessibilityText
+        }
+    }
+
+    @ViewBuilder
+    private var dragIndicator: some View {
+        Button(action: dragIndicatorAction) {
+            RoundedRectangle(cornerRadius: 2)
+                .fill(Color.borderMedium)
+                .frame(width: 36, height: 4)
```

Comment:
```suggestion
                .frame(width: 36, height: 4)
                .frame(height: Image.defaultIconSize)
```
This adds an extended vertical taparea, otherwise its really hard to tap the bar itself